server/chat-commands/info.ts

```diff
@@ -2866,9 +2866,93 @@ export const commands: Chat.ChatCommands = {
 			);
 		}
 	},
+	altlog: 'altslog',
+	altslog(target, room, user) {
+		this.checkCan('lock');
+		target = toID(target);
+		if (!target) {
+			return this.parse(`/help altslog`);
+		}
+		return this.parse(`/join view-altslog-${target}`);
+	},
+	altsloghelp: [
+		`/altslog [userid] - View the alternate account history for the given [userid]. Requires: % @ &`,
+	],
+};
+
+export const handlers: Chat.Handlers = {
+	onRename(user, oldID, newID) {
+		if (oldID === newID || !Config.usesqlite || [oldID, newID].some(f => f.startsWith('guest'))) return;
+		void Chat.database.run(
+			`REPLACE INTO alts_log (to_id, from_id, ip) VALUES (?, ?, ?)`,
+			[newID, oldID, user.latestIp]
+		);
+	},
 };
 
 export const pages: Chat.PageTable = {
+	async altslog(query, user) {
+		this.checkCan('lock');
+		this.title = '[Alts Log]';
+		const target = toID(query.shift());
+		if (!target) {
+			return this.errorReply(`Please specify a user to find alternate accounts for.`);
+		}
+		this.title += ` ${target}`;
+		if (!Config.usesqlite) {
+			return this.errorReply(`The alternate account log is current disabled.`);
+		}
+		const rawLimit = query.shift() || "100";
+		const num = parseInt(rawLimit);
+		if (num > 3000) {
+			return this.errorReply(`3000 is the maximum number of results from the alternate account log.`);
+		}
+		if (isNaN(num)) {
+			return this.errorReply(`The max results must be a real number (received "${rawLimit}")`);
+		}
```

Comment:
Should you check for negative numbers?

Reply:
...Maybe? This is only realistically gonna be used in a button and SQLite _does_ technically support negative limits. There's no real harm if we don't check - dunno what to tell anyone going out of their way to handicap themselves.
There's also no real reason not to check, so it pretty much comes down to preference. I have no opinion either way.

Reply:
A negative LIMIT will make SQLite return all results, which defeats the max of 3000.

Reply:
Does it? Fascinating. Will swap, then.

Reply:
[SQLite docs](https://www.sqlite.org/lang_select.html#limitoffset):
> If the LIMIT expression evaluates to a negative value, then there is no upper bound on the number of rows returned.